File: frontend/src/types/dailyReview.ts
export interface DailyReviewOverview {
  recommended_count: number
  recent_memory_count: number
  recent_graph_added_count: number
  active_topics: string[]
}

export interface DailyReviewReason {
  code: string
  label: string
}

export interface DailyReviewCard {
  id: string
  title: string
  summary: string
  content: string
  created_at: string | null
  updated_at: string | null
  graph_status: string
  graph_added_at: string | null
  score: number
  tags: string[]
  reasons: DailyReviewReason[]
}

export interface DailyReviewTopic {
  topic: string
  count: number
  last_seen_at: string | null
  summary: string
}

export interface DailyReviewResponse {
  overview: DailyReviewOverview
  recommended: DailyReviewCard[]
  topic_focuses: DailyReviewTopic[]
  graph_highlights: DailyReviewCard[]
  needs_refinement: DailyReviewCard[]
}

File: backend/app/schemas/daily_review.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class DailyReviewOverview(BaseModel):
    recommended_count: int = 0
    recent_memory_count: int = 0
    recent_graph_added_count: int = 0
    active_topics: list[str] = Field(default_factory=list)


class DailyReviewReason(BaseModel):
    code: str
    label: str


class DailyReviewCard(BaseModel):
    id: str
    title: str
    summary: str
    content: str
    created_at: datetime | None = None
    updated_at: datetime | None = None
    graph_status: str = 'not_added'
    graph_added_at: datetime | None = None
    score: int = 0
    tags: list[str] = Field(default_factory=list)
    reasons: list[DailyReviewReason] = Field(default_factory=list)


class DailyReviewTopic(BaseModel):
    topic: str
    count: int = 0
    last_seen_at: datetime | None = None
    summary: str


class DailyReviewResponse(BaseModel):
    overview: DailyReviewOverview
    recommended: list[DailyReviewCard] = Field(default_factory=list)
    topic_focuses: list[DailyReviewTopic] = Field(default_factory=list)
    graph_highlights: list[DailyReviewCard] = Field(default_factory=list)
    needs_refinement: list[DailyReviewCard] = Field(default_factory=list)
